babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -1583,3 +1583,49 @@ SqlNode SqlSelectTopN(SqlParserPos pos) :
             SqlLiteral.createBoolean(withTies, pos));
     }
 }
+
+SqlNode SqlHexCharStringLiteral() :
+{
+    final NlsString p;
+    final String formatString;
+    final String charSet;
+    CharacterSet charSetEnum = null;
+    final HexCharLiteralFormat format;
+}
+{
+    (
+        <PREFIXED_HEX_STRING_LITERAL>
+        {
+            charSet = SqlParserUtil
+                      .trim(SqlParserUtil.getCharacterSet(token.image), " ")
+                      .toUpperCase();
+            if (charSet.equals("LATIN")) {
+                charSetEnum = CharacterSet.LATIN;
+            } else if (charSet.equals("UNICODE")) {
+                charSetEnum = CharacterSet.UNICODE;
+            } else if (charSet.equals("GRAPHIC")) {
+               charSetEnum = CharacterSet.GRAPHIC;
+            } else if (charSet.equals("KANJISJIS")) {
+              charSetEnum = CharacterSet.KANJISJIS;
+            } else {
+              throw SqlUtil.newContextException(getPos(),
+              RESOURCE.unknownCharacterSet(charSet));
```

Comment:
nit: The indentation on this line looks off.

Reply:
Thank you for pointing it out. Due to a larger refactoring at https://github.com/googleinterns/calcite/pull/155/commits/eb9c2fd414f98e835b3ef2d007f0093d7df4e3d9, this issue is outdated. The indentation level in the new file is corrected.